base/compiler/tfuncs.jl

```diff
@@ -2393,7 +2393,7 @@ function setglobal!_nothrow(argtypes::Vector{Any})
     end
     M, s, newty = argtypes
     if M isa Const && s isa Const
-        M, s = M.val, s.val
+        M, s = M.val::Module, s.val::Symbol
```

Comment:
do we know this is true, or do we need to check it?

Reply:
I thought we have some prior check, but it doesn't seem to exist. We need type checks here.